drivers/dac/ad5791/ad5791.c

```diff
@@ -323,7 +324,9 @@ int32_t ad5791_setup(struct ad5791_dev *dev,
 	uint32_t new_ctrl = 0;
 	uint32_t val;
 	int32_t status = 0;
+	uint32_t rbuf_mask;
 
+	rbuf_mask = dev->rbuf_gain2 & AD5791_CTRL_RBUF_MASK;
```

Comment:
`AD5791_CTRL_RBUF_MASK` is defined as `NO_OS_BIT(1)` and `rbuf_gain2` will either have the value 0 or 1, so `rbuf_mask` will always be 0. I think you need to shift `rbuf_gain2` before &.